modules/install/pages/upgrade-cluster-online-full-capacity.adoc

```diff
@@ -0,0 +1,236 @@
+(Prior to installation, if necessary, remove any old configuration files from `/opt/couchbase`, using the command `rm -r /opt/couchbase`.)
+
+* During the cluster-upgrade process, _swap_ nodes and _upgraded_ nodes will be introduced to the cluster by the _node-addition_ procedure; while _failed over spare nodes_ and _nodes to be upgraded_ will be withdrawn from the cluster by means of the _node removal_ and _swap rebalance_ procedures.
++
+For information on node-addition, see xref:learn:clusters-and-availability/nodes.adoc#clusters[Clusters].
+For information on node-removal, see xref:learn:clusters-and-availability/removal.adoc[Removal].
+For information on _swap rebalance_, see xref:install:upgrade-strategies.adoc#swap-rebalance[Swap Rebalance].
+
+* Nodes will be upgraded _one at a time_.
+
+The overall procedure is described in seven stages, below.
+
+[#prepare-a-spare-node]
+== Stage One: Prepare a Spare Node
+
+Install the latest version of Couchbase Server on the spare node.
+For instructions, see xref:install:install-intro.adoc[Install]; and follow steps 1, 2, and 3.
+
+[#prepare-the-cluster]
+== Stage Two: Prepare the Cluster
+
+Back up all cluster-data.
+This can be performed either with xref:backup-restore:enterprise-backup-restore.adoc[cbbackupmgr] or with the xref:learn:services-and-indexes/services/backup-service.adoc[Backup Service]; and should be a _full_ (rather than an incremental) backup.
+
+For example, to use `cbbackupmgr` to configure an archive and repository for the backup, a command of the following form should be entered:
+
+[source,bash]
+----
+cbbackupmgr config --archive ${ABS_PATH_TO_ARCHIVE} --repo ${REPO_NAME}
+----
+
+Here, `ABS_PATH_TO_ARCHIVE` is an absolute path to a filesystem location that will serve as the archive within which the backed up data will reside.
+The `REPO_NAME` is the name of the repository that will be associated with the location.
+
+Once the archive and repository have been created, a command such as the following performs a full backup:
+
+[source,bash]
+----
+cbbackupmgr backup --archive ${ABS_PATH_TO_ARCHIVE} --repo ${REPO_NAME} \
+--cluster ${CLUSTER_ADDRESS} --username ${USERNAME} --password ${PASSWORD} \
+--full-backup
+----
+
+Here, the `CLUSTER_ADDRESS` is the IP address or domain name of the cluster that is being backed up.
+The `full-backup` flag ensures that the backup is indeed a _full_ backup.
+
+For the equivalent procedure as performed by the Backup Service, see xref:manage:manage-backup-and-restore/manage-backup-and-restore.adoc#run-an-immediate-backup[Run an Immediate Backup].
+
+In the event of node-upgrade failing, the backed-up data can be restored by means of xref:backup-restore:cbbackupmgr-restore.adoc[cbbackupmgr restore] or xref:cli:cbtools/cbrecovery.adoc[cbrecovery]; or by means of the Backup Service (see xref:manage:manage-backup-and-restore/manage-backup-and-restore.adoc#restore-backups[Restore Backups]).
```

Comment:
Looking at this, I think it would be better to remove this paragraph (which seems partly incorrect and partly incomplete), and open a ticket for a fuller subsection, entitled something like Data Recovery After Upgrade-Failure. 
Then we can go into more detail about each of these, and how it should be used. What do you think?

Reply:
I agree, I think it would be better for us to cover this more comprehensively since there's a fair few options (all with pros/cons along with versioning to be aware of).